exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -54,9 +54,22 @@
 ExportServiceRequestT = TypeVar("ExportServiceRequestT")
 ExportResultT = TypeVar("ExportResultT")
 
+_ENVIRON_TO_COMPRESSION = {
+    None: None,
+    "gzip": Compression.Gzip,
+    "deflate": Compression.Deflate,
+}
 
-class OTLPCompression(enum.Enum):
-    gzip = "gzip"
+
+def environ_to_compression(environ_key: str) -> Optional[Compression]:
+    environ_value = environ.get(environ_key)
+    if environ_value not in _ENVIRON_TO_COMPRESSION:
+        raise Exception(
```

Comment:
Hmm im not sure. I was thinking since this runs as startup code, it would make debugging easier for the user i.e. [2. from your link](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#user-content-basic-error-handling-principles:~:text=The%20API%20or%20SDK%20may%20fail,config%20settings%20received%20from%20the%20Collector.):
> The API or SDK may fail fast and cause the application to fail on initialization, e.g. because of a bad user config or environment, but MUST NOT cause the application to fail later at run time, e.g. due to dynamic config settings received from the Collector.

I'm fine with either way, what do others think?

Reply:
I think exception is better since this is "setup" code and not instrumentation.

Reply:
nit: perhaps have a subclass Exception class for this?

Reply:
@owais is there any existing exception for invalid OTel env var values that can't be parsed?

Reply:
fixed

Reply:
I don't remember seeing one. Just thinking someone might wanna catch an otel related exception and handle it differently so perhaps a specific class would be nice but it's not a big deal for me,